src/main/java/teammates/ui/controller/InstructorFeedbackResultsPageData.java

```diff
@@ -25,94 +53,1321 @@
+                if (!responsesGroupedByTeam.get(team).containsKey(question)) {
+                    continue;
+                }
+                
+                List<FeedbackResponseAttributes> responsesGivenTeamAndQuestion = responsesGroupedByTeam.get(team).get(question);
+        
+                InstructorResultsQuestionTable statsTable = buildQuestionTableWithoutResponseRows(question, 
+                                                                               responsesGivenTeamAndQuestion,
+                                                                               "");
+                statsTable.setCollapsible(false);
+                
+                if (!statsTable.getQuestionStatisticsTable().isEmpty()) {
+                    statisticsTablesForTeam.add(statsTable);
+                }
+            }
+        
+            InstructorResultsQuestionTable.sortByQuestionNumber(statisticsTablesForTeam);
+            teamToStatisticsTables.put(team, statisticsTablesForTeam);
+        }
+        
+        sectionPanel.setTeamStatisticsTable(teamToStatisticsTables);
+    }
+
+    private void prepareHeadersForTeamPanelsInSectionPanel(
+                                    InstructorFeedbackResultsSectionPanel sectionPanel) {
+        switch (viewType) {
+            case GIVER_QUESTION_RECIPIENT:
+                sectionPanel.setStatisticsHeaderText("Statistics for Given Responses");
+                sectionPanel.setDetailedResponsesHeaderText("Detailed Responses");
+                break;
+            case RECIPIENT_QUESTION_GIVER:
+                sectionPanel.setStatisticsHeaderText("Received Responses Statistics");
+                sectionPanel.setDetailedResponsesHeaderText("Detailed Responses");
+                break;
+            default:
+                sectionPanel.setDetailedResponsesHeaderText("");
+        }
+    }
+
+    
+    private InstructorResultsQuestionTable buildQuestionTableAndResponseRows(
+                                    FeedbackQuestionAttributes question,
+                                    List<FeedbackResponseAttributes> responses,
+                                    String additionalInfoId) {
+        return buildQuestionTableAndResponseRows(
+                                        question, responses,
+                                        additionalInfoId, 
+                                        null, true);   
+    }
+    
+    /**
+     * Builds question tables without response rows, but with stats
+     * @param question
+     * @param responses  responses to compute statistics for
+     * @param additionalInfoId
+     */
+    private InstructorResultsQuestionTable buildQuestionTableWithoutResponseRows(
+                                    FeedbackQuestionAttributes question,
+                                    List<FeedbackResponseAttributes> responses,
+                                    String additionalInfoId) {
```

Comment:
<code>buildQuestionTableAndResponseRows</code> is called somewhere else with a non "" value. I think it's preferable to have additionalInfoId as a parameter and eventually we pass in a non "" value as another issue. The "" now causes duplicate html ids on the page.